hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/HoodieParquetFileFormatHelper.scala

```diff
@@ -33,8 +33,9 @@ object HoodieParquetFileFormatHelper {
     val sparkRequestStructFields = requiredSchema.map(f => {
       val requiredType = f.dataType
       if (fileStructMap.contains(f.name) && !isDataTypeEqual(requiredType, fileStructMap(f.name))) {
-        implicitTypeChangeInfo.put(new Integer(requiredSchema.fieldIndex(f.name)), org.apache.hudi.common.util.collection.Pair.of(requiredType, fileStructMap(f.name)))
-        StructField(f.name, fileStructMap(f.name), f.nullable)
+        val readerType = addMissingFields(requiredType, fileStructMap(f.name))
```

Comment:
we should try and write UTs for these instead of e2e. our run tests run time will keep spiking if we end up writing e2e for all combinations.